spec/models/product_importer_spec.rb

```diff
@@ -169,11 +169,28 @@
     end
   end
 
+  describe "when uploading a spreadsheet with some malformed data" do
+    # Use a simple string as CSV.generate will do some escaping
+    let(:csv_data) {
+      csv = "name,producer,category,on_hand,price,units,unit_type,shipping_category\n"
+      csv += "Good Carrots,#{enterprise.name},Vegetables,5,3.20,500,g,#{shipping_category.name}\n"
+      csv += "Malformed \rBrocolli,#{enterprise.name},Vegetables,8,2.50,200,g,#{shipping_category.name}\n"
+    }
+    let(:importer) { import_data csv_data }
+
+    # an unquoted \n will create a non valid line which will fail entry validation hence why we are only testing with \r
```

Comment:
oh, I got it now!
If you add a \n in the middle of a line it will create a non valid line :+1: 
I was confused because the \n is there! maybe "an unquoted \n in the middle of a line" could help the next reader :-)